neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -1893,6 +1902,15 @@ object Neo4jStoreSingleton : GraphStore {
                                     error = it)
                         }.bind()
 
+                /* TODO: (kmm) The logic behind using region-code to fetch 'tax-rates' will fail
+                         when customers are linked to multiple regions. Currently it is assumed
+                         that a customer belongs to only one region. */
+                val region = customerStore.getRelated(customerId, customerRegionRelation, transaction)
```

Comment:
RegionCode can be part of Plan properties. Then it will work even if the customer is linked with multiple regions.  And then optionally we can just check if the customer belongs to the region.  Optionally because customer cannot by the product plan if it is not in the proper segment. And we put the customer in the valid segment only after it is approved for a region.